packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -350,6 +350,32 @@ class _DevFSHttpWriter {
   }
 }
 
+// Basic statistics for DevFS update operation.
+class UpdateFSReport {
+  UpdateFSReport({bool success = false,
+    int invalidatedSourcesCount = 0, int syncedBytes = 0}) {
+    _success = success;
+    _invalidatedSourcesCount = invalidatedSourcesCount;
+    _syncedBytes = syncedBytes;
+  }
+
+  bool get success => _success;
+  int get invalidatedSourcesCount => _invalidatedSourcesCount;
+  int get syncedBytes => _syncedBytes;
+
+  void incorporateResults(UpdateFSReport report) {
+    if (!report._success) {
+      _success = false;
+    }
+    _invalidatedSourcesCount += report._invalidatedSourcesCount;
+    _syncedBytes += report._syncedBytes;
```

Comment:
if (report._sucess) is false it seems to still go and update the remaining two fields and also _success is not set to true when report._success is true (it assumes the value was true which is correct in the code below but the default value of the constructor is false, maybe the default value should be true).

the code here could be
if (!report._sucess) {
  _success = false;
} else {
  _invalidatedSourceCount += ...
  _syncedBytes += ...
}

Reply:
Not sure how useful that is to report stats on a failed operation.

Reply:
Here we just count and faithfully make aggregated results available for consumer to report them if they decide it's useful.
When synchronization failed, I can see how number of transferred bytes can be interesting to know especially if it's large number.